frontend/ui/widget/inputtext.lua

```diff
@@ -141,8 +143,90 @@ if Device:isTouchDevice() or Device:hasDPad() then
                 self.text_widget:moveCursorToXY(x, y, true) -- restrict_to_view=true
                 self.charpos, self.top_line_num = self.text_widget:getCharPos()
             end
-            if Device:hasClipboard() and Device.input.hasClipboardText() then
-                self:addChars(Device.input.getClipboardText())
+            -- clipboard dialog
+            if Device:hasClipboard() then
+                if self.do_select then -- select mode on
+                    if self.selection_start_pos then -- select end
+                        Device.input.setClipboardText(table.concat(self.charlist, "", self.selection_start_pos, self.charpos-1))
+                        UIManager:show(Notification:new{
+                            text = _("Text copied to clipboard"),
+                        })
+                        self.selection_start_pos = nil
+                        self.do_select = false
+                        return true
+                    else -- select start
+                        self.selection_start_pos = self.charpos
+                        UIManager:show(Notification:new{
+                            text = _("Hold the end of selection"),
+                        })
+                        return true
+                    end
+                end
+                local input_dialog
+                input_dialog = require("ui/widget/inputdialog"):new{
+                    title = _("Clipboard"),
+                    stop_events_propagation = true,
+                    input_hint = _("empty"),
+                    input = Device.input.getClipboardText(),
+                    buttons = {
+                        {
+                            {
+                                text = _("All"),
```

Comment:
A bit not obvious that these `All | Line | Word` means `Select all | line | word`. But well, these are short and balanced, so well :)